examples/platform/linux/AppMain.cpp

```diff
@@ -128,30 +128,46 @@ void OnSignalHandler(int signum)
     {
     case SIGVTALRM:
         bootReason = BootReasonType::kPowerOnReboot;
+        Server::GetInstance().DispatchShutDownAndStopEventLoop();
         break;
     case SIGALRM:
         bootReason = BootReasonType::kBrownOutReset;
+        Server::GetInstance().DispatchShutDownAndStopEventLoop();
         break;
     case SIGILL:
         bootReason = BootReasonType::kSoftwareWatchdogReset;
+        Server::GetInstance().DispatchShutDownAndStopEventLoop();
         break;
     case SIGTRAP:
         bootReason = BootReasonType::kHardwareWatchdogReset;
+        Server::GetInstance().DispatchShutDownAndStopEventLoop();
         break;
     case SIGIO:
         bootReason = BootReasonType::kSoftwareUpdateCompleted;
+        Server::GetInstance().DispatchShutDownAndStopEventLoop();
         break;
     case SIGINT:
         bootReason = BootReasonType::kSoftwareReset;
+        Server::GetInstance().DispatchShutDownAndStopEventLoop();
+        break;
+    case SIGUSR1:
+        Server::GetInstance().DispatchSoftwareFaultEvent(SoftwareDiagnostics::Events::SoftwareFault::Id);
+        break;
+    case SIGUSR2:
+        Server::GetInstance().DispatchGeneralFaultEvent(GeneralDiagnostics::Events::HardwareFaultChange::Id);
+        break;
+    case SIGHUP:
+        Server::GetInstance().DispatchGeneralFaultEvent(GeneralDiagnostics::Events::RadioFaultChange::Id);
+        break;
+    case SIGTTIN:
+        Server::GetInstance().DispatchGeneralFaultEvent(GeneralDiagnostics::Events::NetworkFaultChange::Id);
```

Comment:
Please move all fault handling event management *away from signals* and add APIs to Software/General Diagnostics cluster server to trigger logging of said events. We could use a named pipe on Linux to allow receiving single-byte requests for fault events.

Reply:
I will proceed to refactor to implement the event handling APIs in cluster server in this PR. https://github.com/project-chip/connectedhomeip/issues/19114 is created to use pipe instead of signal to trigger the event, need to sync with CSG to update the test plans